src/web/src/components/CourseList.vue

```diff
@@ -128,27 +149,41 @@ export default {
       this.selectedSubsemester = options[0].value;
       return options;
     },
-    /**
-     * Returns a list of courses that match the selected filters
-     * @returns {Course[]}
-     */
-    filteredCourses() {
-      return this.courses.filter(
-        ({ date_start, date_end, department, title, semester }) => {
-          return (
-            (!this.selectedSubsemester ||
-              (this.selectedSubsemester.date_start.getTime() ===
-                date_start.getTime() &&
-                this.selectedSubsemester.date_end.getTime() ===
-                  date_end.getTime())) &&
-            (!this.selectedDepartment ||
-              this.selectedDepartment === department) &&
-            (!this.textSearch ||
-              title.includes(this.textSearch.toUpperCase())) &&
-            this.selectedSemester === semester
-          );
-        }
+    // return a list of the titles of each course.
+    mapCourseNames() {
+      return this.courseList
+        .filter(
+          (course) =>
+            this.selectedDepartment === null ||
+            course.department === this.selectedDepartment
+        )
+        .map((course) => {
+          if (course.full_title !== null) return course.full_title;
+          else return course.title;
+        });
+    },
+    // returns exact match if possible.
+    // if no exact match exists, returns similar options.
+    filterCourses() {
+      let filtered = this.courseList.filter(
+        (course) =>
+          (course.full_title !== null &&
+            course.full_title.toUpperCase() ===
+              this.textSearch.toUpperCase()) ||
+          course.title.toUpperCase() === this.textSearch.toUpperCase()
       );
+      if (
+        filtered.length === 1 &&
+        (this.selectedDepartment === null ||
```

Comment:
move department filtering into `this.courseList.filter` to instead of doing it twice?